core/commonTest/src/kotlinx/io/tests/buffer/BufferPrimitivesTest.kt

```diff
@@ -0,0 +1,309 @@
+        val buffer = PlatformBufferAllocator.allocate(16)
+        buffer.storeUIntAt(0, 1u)
+        buffer.storeUIntAt(8L, 2u)
+        assertEquals(1u, buffer.loadUIntAt(0))
+        assertEquals(1u, buffer.loadUIntAt(0L))
+        assertEquals(2u, buffer.loadUIntAt(8))
+        assertEquals(2u, buffer.loadUIntAt(8L))
+    }
+
+    @Test
+    fun storeCopyAndLoadUInt() {
+        val buffer = PlatformBufferAllocator.allocate(24)
+        buffer.storeUIntAt(0, 1u)
+        buffer.storeUIntAt(8L, 2u)
+        buffer.storeUIntAt(16L, 3u)
+        val copy = PlatformBufferAllocator.allocate(24)
+        buffer.copyTo(copy, 0, 16, 0)
+        assertEquals(1u, buffer.loadUIntAt(0))
+        assertEquals(2u, buffer.loadUIntAt(8))
+        assertEquals(3u, buffer.loadUIntAt(16))
+        assertEquals(1u, copy.loadUIntAt(0))
+        assertEquals(2u, copy.loadUIntAt(8))
+        assertEquals(0u, copy.loadUIntAt(16))
+    }
+
+    @Test
+    fun storeAndLoadULong() {
+        val buffer = PlatformBufferAllocator.allocate(16)
+        buffer.storeULongAt(0, 1u)
+        buffer.storeULongAt(8L, 2u)
+        assertEquals(1u, buffer.loadULongAt(0))
+        assertEquals(1u, buffer.loadULongAt(0L))
+        assertEquals(2u, buffer.loadULongAt(8))
+        assertEquals(2u, buffer.loadULongAt(8L))
+    }
+
+    @Test
+    fun storeCopyAndLoadULong() {
+        val buffer = PlatformBufferAllocator.allocate(24)
+        buffer.storeULongAt(0, 1u)
+        buffer.storeULongAt(8L, 2u)
+        buffer.storeULongAt(16L, 3u)
+        val copy = PlatformBufferAllocator.allocate(24)
+        buffer.copyTo(copy, 0, 16, 0)
+        assertEquals(1u, buffer.loadULongAt(0))
+        assertEquals(2u, buffer.loadULongAt(8))
+        assertEquals(3u, buffer.loadULongAt(16))
+        assertEquals(1u, copy.loadULongAt(0))
+        assertEquals(2u, copy.loadULongAt(8))
+        assertEquals(0u, copy.loadULongAt(16))
+    }
+
+    @Test
+    fun storeAndLoadUShort() {
+        val buffer = PlatformBufferAllocator.allocate(16)
+        buffer.storeUShortAt(0, 1u)
+        buffer.storeUShortAt(8L, 2u)
+        assertEquals(1u, buffer.loadUShortAt(0))
+        assertEquals(1u, buffer.loadUShortAt(0L))
+        assertEquals(2u, buffer.loadUShortAt(8))
+        assertEquals(2u, buffer.loadUShortAt(8L))
+    }
+
+    @Test
+    fun storeCopyAndLoadUShort() {
+        val buffer = PlatformBufferAllocator.allocate(24)
+        buffer.storeUShortAt(0, 1u)
+        buffer.storeUShortAt(8L, 2u)
+        buffer.storeUShortAt(16L, 3u)
+        val copy = PlatformBufferAllocator.allocate(24)
+        buffer.copyTo(copy, 0, 16, 0)
+        assertEquals(1u, buffer.loadUShortAt(0))
+        assertEquals(2u, buffer.loadUShortAt(8))
+        assertEquals(3u, buffer.loadUShortAt(16))
+        assertEquals(1u, copy.loadUShortAt(0))
+        assertEquals(2u, copy.loadUShortAt(8))
+        assertEquals(0u, copy.loadUShortAt(16))
+    }
+
+// Test generator (works only on JVM):
```

Comment:
Consider extracting this to file with the `main` function.

Reply:
What's the point?
It should be in the same file (then it cannot be uncommented because it is a common test and this generator uses reflection, otherwise no one will be able to find it